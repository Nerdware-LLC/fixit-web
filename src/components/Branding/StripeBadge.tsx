import { styled } from "@mui/material/styles";
import Tooltip, { type TooltipProps } from "@mui/material/Tooltip";
import { Anchor, type AnchorProps } from "@/components/Navigation/Anchor";
import StripeBadgeSVG from "@/images/powered_by_Stripe_purple.svg";
import { brandingClassNames } from "./classNames";
import type { Simplify, Except } from "type-fest";

export const StripeBadge = ({
  href = STRIPE_LINKS.LANDING_PAGE,
  tabIndex = -1,
  tooltipProps: { title, ...tooltipProps } = {},
  anchorProps = {},
  ...imgProps
}: StripeBadgeProps) => (
  <Tooltip title={title || href} {...tooltipProps}>
    <StyledAnchor
      href={href}
      target="_blank"
      rel="noreferrer"
      tabIndex={tabIndex}
      className={brandingClassNames.stripeBadgeAnchor}
      {...anchorProps}
    >
      <img
        src={StripeBadgeSVG}
        alt="Stripe Badge"
        className={brandingClassNames.stripeBadgeImg}
        {...imgProps}
      />
    </StyledAnchor>
  </Tooltip>
);

/**
 * A collection of links to Stripe's website.
 */
export const STRIPE_LINKS = {
  LANDING_PAGE: "https://stripe.com/",
  CONNECT_ACCOUNT_AGREEMENT: "https://stripe.com/connect-account/legal/full",
} as const satisfies Record<string, string>;

const StyledAnchor = styled(Anchor)({
  display: "contents",

  [`& > .${brandingClassNames.stripeBadgeImg}`]: {
    height: "100%",
    minHeight: "1.5rem !important", // smaller than this, and the text is too small on most screens
    maxHeight: "3rem !important", // currently no reason the badge should ever be taller than this
    objectFit: "contain",
  },
});

export type StripeBadgeProps = {
  href?: (typeof STRIPE_LINKS)[keyof typeof STRIPE_LINKS];
  tooltipProps?: Simplify<Partial<Except<TooltipProps, "children" | "className">>>;
  anchorProps?: Simplify<Except<AnchorProps, "ref" | "href" | "target" | "rel" | "className">>;
} & Except<React.ComponentPropsWithoutRef<"img">, "src" | "className" | "children">;
