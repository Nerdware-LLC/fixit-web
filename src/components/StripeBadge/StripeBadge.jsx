import React from "react";
import StripeBadgeSrc from "../../images/powered_by_stripe.png";
import { oneOf } from "../../types";

export const StripeBadge = ({ size = "normal", ...props }) => (
  <span {...props}>
    <a href="https://stripe.com/">
      <img src={StripeBadgeSrc} alt={"Stripe Badge"} />
    </a>
  </span>
);

StripeBadge.propTypes = {
  size: oneOf(["small", "normal", "large"])
};
