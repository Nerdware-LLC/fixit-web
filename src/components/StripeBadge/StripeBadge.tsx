import StripeBadgeSVG from "@images/powered_by_Stripe_purple.svg";

export const StripeBadge = ({
  href = "https://stripe.com/",
  style,
  ...props
}: StripeBadgeProps) => (
  <a
    href={href}
    className={`${stripeBadgeClassNames.root} ${stripeBadgeClassNames.anchor}`}
    target="_blank"
    rel="noreferrer"
    style={style}
  >
    <img
      src={StripeBadgeSVG}
      className={stripeBadgeClassNames.img}
      alt="Stripe Badge"
      style={style}
      {...props}
    />
  </a>
);

export const stripeBadgeClassNames = {
  root: "stripe-badge-root",
  anchor: "stripe-logo-anchor",
  img: "stripe-logo-img",
};

export type StripeBadgeProps = {
  href?: string;
} & React.ComponentPropsWithoutRef<"img">;
