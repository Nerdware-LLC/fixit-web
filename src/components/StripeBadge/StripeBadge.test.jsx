import renderer from "react-test-renderer";
import { StripeBadge } from "./StripeBadge";

it("renders correctly", () => {
  const tree = renderer.create(<StripeBadge />).toJSON();
  expect(tree).toMatchSnapshot();
});
