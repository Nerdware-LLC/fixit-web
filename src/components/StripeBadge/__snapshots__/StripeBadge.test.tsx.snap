// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<a
  className="stripe-badge-root stripe-logo-anchor"
  href="https://stripe.com/"
  rel="noreferrer"
  target="_blank"
>
  <img
    alt="Stripe Badge"
    className="stripe-logo-img"
    src="/src/images/powered_by_Stripe_purple.svg"
  />
</a>
`;
